src/odemis/driver/simstreakcam.py

```diff
@@ -0,0 +1,516 @@
+
+    def _getNewImage(self):
+        """Gets a new image from the list of fake images provided."""
+        # image counter to provide different images to live view
+        if not hasattr(self, "_img_counter"):
+            self._img_counter = 0  # initialize the image counter
+        self._img_counter = self._img_counter % len(self._img_list)
+        image = self._img_list[self._img_counter]
+        self._img_counter += 1
+        return image
+
+    def _generate(self):
+        """
+        Generates the fake output based on the resolution.
+        """
+        gen_img = self._getNewImage()
+        gen_img = self._simulate(gen_img)
+        timer = self._generator  # might be replaced by None afterwards, so keep a copy
+        self._waitSync()
+        if self._sync_event:
+            # If sync event, we need to simulate period after event (not efficient, but works)
+            time.sleep(self.exposureTime.value)
+
+        # update the trigger rate VA and MD for the current image
+        self.parent._delaybox._updateTriggerRate()
+
+        metadata = gen_img.metadata.copy()  # MD of image
+        metadata.update(self._metadata)  # MD of camera
+        self._mergeMetadata(metadata)
+
+        # update fake output metadata
+        exp = timer.period
+        metadata[model.MD_ACQ_DATE] = time.time() - exp
+        metadata[model.MD_EXP_TIME] = exp
+        metadata.pop(model.MD_PIXEL_SIZE, None)  # TODO get rid of this MD for now
+
+        # TODO check if that time list can be used to update the live view legend if there is one
+        if self.parent._streakunit.streakMode.value:  # if in Mode 'Operate'
+            # shape of gen_image is (time, lambda)
+            timeRange = self.parent._streakunit.timeRange.value  # time range of the time axis in image
+            # each px in time axis corresponds to a time stamp (easy approch: equally distributed)
+            step_time = timeRange/float(gen_img.shape[0])
+            metadata[model.MD_TIME_LIST] = list(numpy.arange(0, timeRange, step_time))  # append fake time list
+        logging.debug("Generating new fake image of shape %s", gen_img.shape)
+
+        img = model.DataArray(gen_img, metadata)
+
+        # send the new image (if anyone is interested)
+        self.data.notify(img)
+
+        # simulate exposure time
+        timer.period = self.exposureTime.value
+
+    def _waitSync(self):
+        """
+        Block until the Event on which the dataflow is synchronised has been
+          received. If the DataFlow is not synchronised on any event, this
+          method immediately returns
+        """
+        if self._sync_event:
+            self._evtq.get()
+
+    def _simulate(self, img):
+        """
+        Processes the fake image based on resolution and binning.
+        """
+        binning = self.binning.value
+        res = self.resolution.value
+        sim_img = img.reshape((res[1], binning[0], res[0], binning[1])).mean(axis=3).mean(axis=1)
```

Comment:
I like the "1 single line" to compute 2D binning :-)
`.mean()` always converts the data to float64. To simulate better, you should convert back to the original dtype. -> `astype(img.dtype)`.

Reply:
thanks :D